ReactWindows/ReactNative.Shared/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,229 @@
+using ReactNative.UIManager;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation;
+using Windows.UI.Xaml.Automation.Peers;
+
+namespace ReactNative.Common
+{
+    /// <summary>
+    /// Accessibility helper methods.
+    /// </summary>
+    public static class AccessibilityHelper
+    {
+        #region ImportantForAccessibility
+
+        /// <summary>
+        /// Sets the ImportantForAccessibility property for uiElement.
+        /// It uses AutomationProperties.AccessibilityView to implement the element and its children
+        /// visibility to narrator. ImportantForAccessibility value is stored as an attached property.
+        /// </summary>
+        /// <param name="uiElement"></param>
+        /// <param name="importantForAccessibilityNew"></param>
+        public static void SetImportantForAccessibility(UIElement uiElement, ImportantForAccessibility importantForAccessibilityNew)
+        {
+            var importantForAccessibilityOld = GetImportantForAccessibilityAttached(uiElement);
+            if (importantForAccessibilityOld == importantForAccessibilityNew)
+            {
+                return;
+            }
+
+            SetImportantForAccessibilityAttached(uiElement, importantForAccessibilityNew);
+
+            var uiElementAutomationPeer = FrameworkElementAutomationPeer.FromElement(uiElement);
+            // If this element is 'hidden' by a parrent, do not update AccessibilityView property.
+            if (IsHiddenElement(uiElementAutomationPeer))
+            {
+                return;
+            }
+
+            // Update AccessibilityView property accordingly.
+            switch (importantForAccessibilityNew)
+            {
+                case ImportantForAccessibility.Auto:
+                    uiElement.ClearValue(AutomationProperties.AccessibilityViewProperty);
```

Comment:
>ClearValue [](start = 30, length = 10)

this method seems to "set" the attached property. So if user sets to Auto she will expect to get Auto when she does Get later. But if we use ClearValue here then the value returned by Get may be influenced by all the mechanism of a DependencyProperty effective value determination, and so may differ from the default value specified in attached property registration. I think we should not use ClearValue here since it's a bug.

Reply:
No this deals with AccessibilityViewProperty (that can be Raw, Control or Content). Auto/Yes/No/NoHideDesc is something that is set by SetImportantForAccessibilityAttached above.  The purpose for "ClearValue" usage is to make sure that when we want to set ImportantForAccessibility to Auto, we get whatever is default for the control AccessibilityView property (either Raw/Control or Content). Removing the attached property will result in default from Automation Peer being used.

---
In reply to: [163119782](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163119782) [](ancestors = 163119782)